spark-cassandra-connector/src/it/scala/org/apache/spark/sql/cassandra/CassandraSQLSpec.scala

```diff
@@ -314,8 +321,264 @@ class CassandraSQLSpec extends SparkCassandraITFlatSpecBase {
       session.execute("insert into sql_test.timestamp_conversion_bug (k, v, d) values (1, 3, '2015-01-03 17:13')")
       session.execute("insert into sql_test.timestamp_conversion_bug (k, v, d) values (1, 4, '2015-01-03 18:13')")
     }
-    val cc = new CassandraSQLContext(sc)
-    cc.setKeyspace("sql_test")
     cc.cassandraSql("select k, min(d), max(d) from timestamp_conversion_bug group by k").collect()
   }
+
+  it should "allow to insert null" in {
+    conn.withSessionDo { session =>
+      session.execute("create table sql_test.null_test (k int, v int, d text, primary key(k,v))")
+      session.execute("insert into sql_test.null_test (k, v, d) values (1, 1, 'one')")
+      session.execute("insert into sql_test.null_test (k, v, d) values (1, 2, 'two')")
+      session.execute("insert into sql_test.null_test (k, v) values (1, 3)")
+      session.execute("create table sql_test.null_test2 (k int, v int, d text, primary key(k,v))")
+    }
+
+    cc.cassandraSql("select * from null_test where d = 'one' ").collect() should have length 1
+    cc.cassandraSql("select * from null_test where d IS NULL ").collect() should have length 1
+    cc.cassandraSql("select * from null_test where d IS NOT NULL ").collect() should have length 2
+    cc.cassandraSql("select * from null_test where v > 1 ").collect() should have length 2
+    cc.cassandraSql("INSERT OVERWRITE TABLE null_test2 SELECT k, v, d FROM null_test")
+    cc.cassandraSql("select * from null_test2").collect() should have length 3
+  }
+
+  it should "allow to query on table with upper case names" in {
+    conn.withSessionDo { session =>
+      session.execute("create table sql_test.\"Upper_Case_Table\"(" +
+        "\"KEY\" int, \"VALUE\" int, d text, primary key(\"KEY\",\"VALUE\"))")
+      session.execute("insert into sql_test.\"Upper_Case_Table\"(\"KEY\",\"VALUE\", d) values (1, 1, 'one')")
+      session.execute("insert into sql_test.\"Upper_Case_Table\"(\"KEY\",\"VALUE\", d) values (1, 2, 'two')")
+      session.execute("insert into sql_test.\"Upper_Case_Table\"(\"KEY\",\"VALUE\", d) values (1, 3, 'three')")
+    }
+    cc.cassandraSql("select KEY, VALUE from Upper_Case_Table where VALUE > 1").collect() should have length 2
+  }
+
+  it should "not find non-exist tables" in {
+    an [ExecutionException] should be thrownBy {
+      cc.tableExists(TableIdent("non_exist", "non_exist"))
+    }
```

Comment:
ExecutionException is a RuntimeException, so it's not caught by the method, Boolean should return.